strawberryfields/backends/states.py

```diff
@@ -814,6 +815,43 @@ def expand_dims(op, n, modes):
 
         return mean, var
 
+    def number_expectation(self, modes):
+        """
+        Calculates the expectation value of the product of the number operators of the modes.
+        Args:
+            modes (list): list of modes for which one wants the expectation of the product of their number operator.
+        Return:
+            (float): the expectation value.
+        """
```

Comment:
Note: it could be good to add a small example here. For instance:
```rest
**Example**

Consider the following program:

.. code-block:: python

    prog = sf.Program(3)

    with prog.context as q:
        ops.Sgate(0.5) | q[0]
        ops.Sgate(0.5) | q[1]
        ops.Sgate(0.5) | q[2]
        ops.BSgate(np.pi/3, 0.1) |  (q[0], q[1])
        ops.BSgate(np.pi/3, 0.1) |  (q[1], q[2])

Executing this on the Fock backend,

>>> eng = sf.Engine("fock", backend_options={"cutoff_dim": 10})
>>> state = eng.run(prog).state

we can compute the expectation value :math:`\langle \hat{n}_0\hat{n}_2\rangle`:

>>> state.number_expectation([0, 2])
2.423234
```
```

Reply:
Done.